Buildings/Experimental/OpenBuildingControl/CDL/Discrete/DayType.mo

```diff
@@ -0,0 +1,104 @@
+within Buildings.Experimental.OpenBuildingControl.CDL.Discrete;
+model DayType "Block that outputs a signal that indicates week-day or week-end"
+  parameter Integer nout = 2
+    "Number of days to output. Set to two for one day predictions";
+  parameter Types.Day[:] days={
+    Types.Day.WorkingDay,
+    Types.Day.WorkingDay,
+    Types.Day.WorkingDay,
+    Types.Day.WorkingDay,
+    Types.Day.WorkingDay,
+    Types.Day.NonWorkingDay,
+    Types.Day.NonWorkingDay}
+    "Array where each element is a day indicator";
+   parameter Integer iStart(min=1, max=size(days, 1)) = 1
+    "Index of element in days at simulation start";
+
+  Interfaces.DayTypeOutput y[nout]
+    "Type of the day for the current and the next (nout-1) days" annotation (
+      Placement(transformation(extent={{100,-10},{120,10}}), iconTransformation(
+          extent={{100,-10},{120,10}})));
+
+protected
+  parameter Modelica.SIunits.Time samplePeriod=86400
+    "Sample period of the component";
+  output Integer iDay(min=1, max=size(days, 1))
+    "Pointer to days that determines what day type is sent to the output";
+  parameter Modelica.SIunits.Time firstSample(fixed=false)
+    "Time when the sampling starts";
+  output Boolean sampleTrigger "True, if sample time instant";
+  output Boolean skipIDayIncrement
+    "If true, don't increment iDay in first sample";
+
+initial equation
+  iDay = iStart;
+  firstSample = ceil(time/86400)*86400;
+  // skipIDayIncrement is true if the simulation starts at midnight.
+  skipIDayIncrement = abs(firstSample-time) < 1E-8;
+equation
+  for i in 1:nout loop
+    y[i] = days[ mod(iDay+i-2, size(days, 1))+1];
+  end for;
+  sampleTrigger = sample(firstSample, samplePeriod);
+  when sampleTrigger then
+    skipIDayIncrement = false;
+    if pre(skipIDayIncrement) then
+      iDay = pre(iDay);
+    else
+      iDay = mod(pre(iDay), size(days, 1))+1;
+    end if;
+  end when;
+  annotation (
+  Icon(coordinateSystem(preserveAspectRatio=true, extent={{-100,
+          -100},{100,100}}), graphics={Rectangle(
+        extent={{-100,-100},{100,100}},
+        lineColor={0,0,127},
+        fillColor={223,211,169},
+        lineThickness=5.0,
+        borderPattern=BorderPattern.Raised,
+        fillPattern=FillPattern.Solid), Text(
+        extent={{-150,150},{150,110}},
+        textString="%name",
+        lineColor={0,0,255})}),
+  defaultComponentName="dayType",
+  Icon(coordinateSystem(preserveAspectRatio=false, extent={{-100,-100},
+            {100,100}}), graphics={Text(
+          extent={{-68,54},{68,-38}},
+          lineColor={0,0,255},
+          textString="day")}),
+    Documentation(info="<html>
+<p>
+This block outputs a periodic signal that indicates the type of the day.
+It can for example be used to generate a signal that indicates whether
+the current time is a work day or a non-working day.
+The output signal is of type
+<a href=\"modelica://Buildings.Experimental.OpenBuildingControl.CDL.Types.Day\">
+Types.Day</a>.
```

Comment:
Please always use the full model name such as `Buildings.Experimental.OpenBuildingControl.CDL.Types.Day` rather than `Types.Day`. The reason is that the rename script will replace the full model name, but not a partial name, and hence the name may not be updated if the library is refactored.